pkg/util/fs_util.go

```diff
@@ -789,11 +789,15 @@ func getSymlink(path string) error {
 func CopyFileOrSymlink(src string, destDir string) error {
 	destFile := filepath.Join(destDir, src)
 	if fi, _ := os.Lstat(src); IsSymlink(fi) {
-		link, err := os.Readlink(src)
+		link, err := EvalSymLink(src)
 		if err != nil {
 			return err
 		}
-		return os.Symlink(link, destFile)
+		linkPath := filepath.Join(destDir, link)
```

Comment:
Could you explain this code a bit? I'm having trouble wraping my head around it.

It looks like `linkPath` is the resolved path of `src` joined with `destDir`. I'm confused how we can be sure this path exists.

Reply:
This was error 1 mentioned in [here](https://github.com/GoogleContainerTools/kaniko/pull/1030#discussion_r376664809)

This ensures, we are copying/creating the required file in kaniko sratch workspace.